cores/esp32/HardwareSerial.cpp

```diff
@@ -97,14 +97,18 @@ void serialEvent2(void) {}
 
 #if !defined(NO_GLOBAL_INSTANCES) && !defined(NO_GLOBAL_SERIAL)
 #if ARDUINO_USB_CDC_ON_BOOT //Serial used for USB CDC
-HardwareSerial Serial0(0);
+// sets UART0 (default console) RX/TX default pins already configured in boot
+HardwareSerial Serial0(0, SOC_RX0, SOC_TX0);
```

Comment:
if this is for UART0 only, would it not be safer to keep the old signature and handle the default pins in the constructor?

Reply:
Yes, this is a better way, less intrusive.